### ISSUE TITLE
linter: `unicorn/prefer-global-this` reports window event-listener method references

### ISSUE THREAD

**Issue by lukpsaxo** [oxc-project/oxc#26597]
### What version of Oxlint are you using?

1.82.0

### What command did you run?

oxlint

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```json
{
    "plugins": ["unicorn"],
    "rules": {
        "unicorn/prefer-global-this": "error"
    }
}
```

### What happened?

Oxlint 1.82.0 reports `window.addEventListener` and `window.removeEventListener` when they are
referenced without being called. `eslint-plugin-unicorn` 74.0.0 treats these as window-specific
APIs when no event name is available and does not report them.

```js
expect(window.addEventListener).toHaveBeenCalled();
```

## Expected

No `unicorn/prefer-global-this` error.
No error is given when calling addEventListener, but when using it like the above sample it _does_ error.

## Actual

One `unicorn/prefer-global-this` diagnostic:

> Prefer `globalThis` over environment-specific global aliases like `window`, `self`, and
> `global`.

The same false positive occurs for a `window.removeEventListener` reference.
